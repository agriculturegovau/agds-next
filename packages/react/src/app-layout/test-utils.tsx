import { Fragment } from 'react';
import { Text } from '../text';
import { NotificationBadge } from '../notification-badge';
import { VisuallyHidden } from '../a11y';
import {
	ChartLineIcon,
	EmailIcon,
	ExitIcon,
	HelpIcon,
	HomeIcon,
	SuccessIcon,
	SettingsIcon,
	FactoryIcon,
	AvatarIcon,
	ChevronsLeftIcon,
	MinusIcon,
} from '../icon';
import {
	DropdownMenuDivider,
	DropdownMenuItem,
	DropdownMenuItemLink,
	DropdownMenuPanel,
	DropdownMenuGroup,
	DropdownMenuGroupLink,
	DropdownMenuItemRadio,
} from '../dropdown-menu';

export const navigationItems = (businessName: string) => [
	{
		options: { disableGroupPadding: true },
		items: [
			{
				label: 'Back to my account',
				icon: ChevronsLeftIcon,
				href: '/account',
			},
		],
	},
	{
		options: { disableGroupPadding: true },
		items: [
			{
				label: (
					<Fragment>
						<Text fontWeight="bold" fontSize="xs">
							{businessName}
						</Text>
						<Text color="muted" fontSize="xs">
							ABN: 00 000 000 000
						</Text>
					</Fragment>
				),
			},
		],
	},
	[
		{
			label: 'Dashboard',
			icon: HomeIcon,
			href: '/',
		},
		{
			label: 'Establishments',
			icon: FactoryIcon,
			href: '/establishments',
			items: [
				{
					label: 'Canberra',
					icon: MinusIcon,
					href: '/establishments/canberra',
				},
				{
					label: 'Sydney',
					icon: MinusIcon,
					href: '/establishments/sydney',
				},
			],
		},
		{
			label: 'Data and Insights',
			icon: ChartLineIcon,
			href: '/intelligence',
		},
		{
			label: 'Compliance',
			icon: SuccessIcon,
			href: '/compliance',
			items: [
				{
					label: 'Audit',
					icon: MinusIcon,
					href: '/compliance/audit',
				},
				{
					label: 'Certificates',
					icon: MinusIcon,
					href: '/compliance/certificates',
				},
			],
		},
	],

	[
		{
			label: 'Help',
			icon: HelpIcon,
			href: '/help',
		},
	],
	[
		{
			label: 'Sign out',
			onClick: console.log,
			icon: ExitIcon,
		},
	],
];

export const exampleData = {
	userNames: {
		short: 'Ab Cd',
		regular: 'Toto Wolff',
		medium: 'Benjamin Harrington',
		long: 'Benjamin Alexander Harrington-Smythe',
	},
	businessNames: {
		short: ['Lorem', 'Ipsum', 'Company'],
		regular: ['Antfix', 'Produce Fresh', 'Organic Co'],
		medium: [
			'Malesuada lacus a tortor blandit fermentum phasellus',
			'Iaculis tortor duis ante nec risus elementum id ui',
			'Purus tortor lacus malesuada phasellus ipsum ex duis libero ante id',
		],
		long: [
			'Duis quis iaculis libero nec consectetur luctus ex in malesuada lacus a tortor blandit fermentum phasellus iaculis ipsum nec purus aliquet id elementum risus duis ante ui',
			'Duis ut a libero duis nisl ex lacus in luctus viverra iaculis iaculis tortor duis malesuada blandit purus diam ipsum ante nec risus elementum id ui nec quis aliquet phasellus fermentum',
			'Blandit iaculis iaculis quis ante diam viverra elementum ui risus nec luctus purus tortor lacus malesuada phasellus ipsum ex duis libero ante id Duis fermentum nisl aliquet duis in a nec ut consectetur',
		],
	},
};

export function ExampleAccountDropdown({
	businesses = exampleData.businessNames.regular,
	onBusinessChange,
	selectedBusinessName,
}: {
	businesses?: string[];
	onBusinessChange: (value: string) => void;
	selectedBusinessName: string;
}) {
	return (
		<DropdownMenuPanel palette="light">
			<DropdownMenuGroup label="Businesses">
				{businesses.map((businessName) => (
					<DropdownMenuItemRadio
						key={businessName}
						checked={businessName === selectedBusinessName}
						secondaryText="ABN 00 000 000 000"
						onClick={() => onBusinessChange?.(businessName)}
					>
						{businessName}
					</DropdownMenuItemRadio>
				))}
				<DropdownMenuGroupLink href="#">View all</DropdownMenuGroupLink>
			</DropdownMenuGroup>
			<DropdownMenuDivider />
			<DropdownMenuGroup label="My account">
				<DropdownMenuItemLink href="/profile" icon={AvatarIcon}>
					Profile
				</DropdownMenuItemLink>
				<DropdownMenuItemLink
					href="/messages"
					icon={EmailIcon}
					endElement={
						<span>
							<NotificationBadge tone="action" value={6} max={99} aria-hidden />
							<VisuallyHidden>, 6 unread</VisuallyHidden>
						</span>
					}
				>
					Messages
				</DropdownMenuItemLink>
				<DropdownMenuItemLink href="/account-settings" icon={SettingsIcon}>
					Account settings
				</DropdownMenuItemLink>
			</DropdownMenuGroup>
			<DropdownMenuDivider />
			<DropdownMenuItem onClick={() => console.log('sign out')} icon={ExitIcon}>
				Sign out
			</DropdownMenuItem>
		</DropdownMenuPanel>
	);
}
