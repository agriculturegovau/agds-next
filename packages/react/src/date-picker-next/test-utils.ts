import * as yup from 'yup';

export const errorMessage = 'Enter a valid date';

// `yup.date()` can sometimes give false positives with certain string values
// Fixes https://github.com/jquense/yup/issues/764
// Tests for this can be viewed in the tests for `DatePicker` and `DateRangePicker`
// IMPORTANT: This is for example purposes only. It should NOT be exported from the design system
export const yupDateField = yup
	.date()
	.transform((current, orig) => (typeof orig === 'string' ? orig : current))
	.typeError(errorMessage);
