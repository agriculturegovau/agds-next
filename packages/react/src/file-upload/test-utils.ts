import { FileStatus, FileWithStatus } from './utils';

export function createExampleFileWithStatus(status?: FileStatus) {
	const bits = ['example'];
	const file: FileWithStatus = new File(bits, 'example.txt', {
		type: 'text/plain',
	});
	file.status = status;
	return file;
}

export function createExampleImageFileWithStatus(status?: FileStatus) {
	const file: FileWithStatus = new File(['example'], 'example.jpg', {
		type: 'image/jpg',
	});
	file.status = status;
	return file;
}
