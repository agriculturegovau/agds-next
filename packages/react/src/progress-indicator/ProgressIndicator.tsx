import { useId } from '../core';
import {
	CollapsingSideBar,
	CollapsingSideBarTitle,
} from '../_collapsing-side-bar';
import { UnorderedList } from '../list';
import { ReplyIcon } from '../icon';
import { Flex } from '../flex';
import { Text } from '../text';
import {
	ProgressIndicatorItem,
	ProgressIndicatorItemButton,
	ProgressIndicatorItemLink,
	ProgressIndicatorItemLinkProps,
} from './ProgressIndicatorItem';
import { ProgressIndicatorList } from './ProgressIndicatorList';

export type ProgressIndicatorProps = {
	/** If the ProgressIndicator is placed on a page with 'bodyAlt' background, please set this to "bodyAlt". */
	background?: 'body' | 'bodyAlt';
	/** The list of items to display. */
	items: ProgressIndicatorItem[];
	/** If true, the `subTitle` above the list of items will not be rendered. */
	hideSubtitle?: boolean;
};

export const ProgressIndicator = ({
	background,
	items,
	hideSubtitle = false,
}: ProgressIndicatorProps) => {
	// The title in this component is not configurable
	const title = 'Progress';

	const subTitle = hideSubtitle
		? undefined
		: `${items.filter(({ status }) => status === 'done').length} of ${
				items.length
		  } steps completed`;

	const hasExplicitActiveDeclared =
		items.filter(({ isActive }) => isActive).length > 0;

	// If no explicit active items are declared, the 'started' status item receives active state by default
	const itemsWithDefaultActive: ProgressIndicatorItem[] =
		hasExplicitActiveDeclared
			? items
			: items.map((item) => ({
					...item,
					isActive: item.status === 'started', // TODO: test this?
			  }));

	return (
		<CollapsingSideBar
			as="section"
			background={background}
			title={<CollapsingSideBarTitle title={title} subtitle={subTitle} />}
			// When `hideTitles` is true, the mobile button label should be "Progress" instead of "x of y steps completed"
			collapseButtonLabel={subTitle || title}
		>
			<ProgressIndicatorList>
				{itemsWithDefaultActive.map(
					({ label, levelTwoItem, ...props }, index) => {
						if (isItemLink(props)) {
							return (
								<ProgressIndicatorItemLink
									key={index}
									background={background}
									{...props}
								>
									{label}
									{levelTwoItem && (
										<Flex
											alignItems="center"
											paddingTop={1}
											gap={0.5}
											css={{ fontWeight: 'normal' }}
										>
											<ReplyIcon color="selected" />
											<Text color="muted" fontSize="xs">
												{levelTwoItem.label}
											</Text>
										</Flex>
									)}
								</ProgressIndicatorItemLink>
							);
						}
						return (
							<ProgressIndicatorItemButton
								key={index}
								background={background}
								{...props}
							>
								{label}
								{levelTwoItem && (
									<Flex
										alignItems="center"
										paddingTop={1}
										gap={0.5}
										css={{ fontWeight: 'normal' }}
									>
										<ReplyIcon color="selected" />
										<Text color="muted" fontSize="xs">
											{levelTwoItem.label}
										</Text>
									</Flex>
								)}
							</ProgressIndicatorItemButton>
						);
					}
				)}
			</ProgressIndicatorList>
		</CollapsingSideBar>
	);
};

export const useProgressIndicatorIds = () => {
	const autoId = useId();
	return {
		buttonId: `progress-indicator-${autoId}-button`,
		bodyId: `progress-indicator-${autoId}-body`,
	};
};

export const isItemLink = (
	item: Omit<ProgressIndicatorItem, 'label'>
): item is ProgressIndicatorItemLinkProps => 'href' in item;
