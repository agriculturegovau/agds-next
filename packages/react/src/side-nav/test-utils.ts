import { SideNavProps } from './SideNav';

export const defaultTestingProps: SideNavProps = {
	title: 'Lodging your tax return',
	titleLink: '/',
	collapseTitle: 'In this section',
	items: [
		{
			href: '/welcome',
			label: 'Welcome',
		},
		{
			href: '/do-you-need-to-lodge-a-tax-return',
			label: 'Do you need to lodge a tax return?',
		},
		{
			href: '/lodge-online',
			label: 'Lodge online',
			items: [
				{
					href: '/lodge-online/pre-filling',
					label: 'Pre-filling your online tax return',
				},
			],
		},
		{
			href: '/whats-new-for-individuals',
			label: "What's new for individuals",
		},
		{
			href: '/why-you-may-receive-a-tax-bill',
			label: 'Why you may receive a tax bill',
		},
		{
			href: '/in-detail',
			label: 'In detail',
			items: [
				{
					href: '/in-detail/record-keeping',
					label: 'Record keeping',
					items: [
						{
							href: '/in-detail/record-keeping/tax',
							label: 'Keeping your tax records',
						},
						{
							href: '/in-detail/record-keeping/incorrect-amounts',
							label: 'Incorrect amounts',
						},
					],
				},
				{
					href: '/tax-receipt',
					label: 'Tax receipt',
				},
				{
					href: '/pre-fill availability',
					label: 'Pre-fill availability',
				},
			],
		},
	],
	activePath: '/in-detail',
};
