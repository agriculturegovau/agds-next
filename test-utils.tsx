import { PropsWithChildren, ReactElement } from 'react';
import { render, RenderOptions } from '@testing-library/react';
import { Core } from '@ag.ds-next/react/core';
import { theme } from '@ag.ds-next/react/ag-branding';

function AllTheProviders({ children }: PropsWithChildren) {
	return <Core theme={theme}>{children}</Core>;
}

const customRender = (
	ui: ReactElement,
	options?: Omit<RenderOptions, 'wrapper'>
) => render(ui, { wrapper: AllTheProviders, ...options });

export * from '@testing-library/react';
export { customRender as render };
