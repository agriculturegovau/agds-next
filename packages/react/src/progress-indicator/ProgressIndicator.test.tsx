import '@testing-library/jest-dom';
import 'html-validate/jest';
import { axe, toHaveNoViolations } from 'jest-axe';
import { render, cleanup } from '../../../../test-utils';
import { ProgressIndicator, ProgressIndicatorProps } from './ProgressIndicator';
import {
	ProgressIndicatorItem,
	ProgressIndicatorItemStatus,
	activeProgressStatusAllowedList,
	allProgressStatuses,
} from './ProgressIndicatorItem';

expect.extend(toHaveNoViolations);

afterEach(cleanup);

function renderProgressIndicator(props: ProgressIndicatorProps) {
	return render(<ProgressIndicator {...props} />);
}

describe('ProgressIndicator', () => {
	describe('With anchors', () => {
		const exampleProps: ProgressIndicatorProps = {
			items: [
				{ href: '#', label: 'Done', status: 'done' },
				{ href: '#', label: 'Blocked', status: 'blocked' },
				{ href: '#', label: 'Doing', status: 'doing' },
				{ href: '#', label: 'Error', status: 'error' },
				{ href: '#', label: 'Saved', status: 'saved' },
				{ href: '#', label: 'Started', status: 'started' },
				{ href: '#', label: 'To do', status: 'todo' },
			],
		};
		it('renders correctly', () => {
			const { container } = renderProgressIndicator(exampleProps);
			expect(container).toMatchSnapshot();
		});
		it('renders valid HTML with no a11y violations', async () => {
			const { container } = renderProgressIndicator(exampleProps);
			expect(container).toHTMLValidate({
				extends: ['html-validate:recommended'],
				rules: {
					'no-inline-style': 'off',
					// react 18s `useId` break this rule
					'valid-id': 'off',
				},
			});
			expect(await axe(container)).toHaveNoViolations();
		});
	});
	describe('With buttons', () => {
		const exampleProps: ProgressIndicatorProps = {
			items: [
				{
					onClick: console.log,
					label: 'Done',
					status: 'done',
				},
				{
					onClick: console.log,
					label: 'Blocked',
					status: 'blocked',
				},
				{
					onClick: console.log,
					label: 'Doing',
					status: 'doing',
				},
				{
					onClick: console.log,
					label: 'Error',
					status: 'error',
				},
				{
					onClick: console.log,
					label: 'Saved',
					status: 'saved',
				},
				{
					onClick: console.log,
					label: 'Started',
					status: 'started',
				},
				{
					onClick: console.log,
					label: 'To do',
					status: 'todo',
				},
			],
		};
		it('renders correctly', () => {
			const { container } = renderProgressIndicator(exampleProps);
			expect(container).toMatchSnapshot();
		});
		it('renders valid HTML with no a11y violations', async () => {
			const { container } = renderProgressIndicator(exampleProps);
			expect(container).toHTMLValidate({
				extends: ['html-validate:recommended'],
				rules: {
					'no-inline-style': 'off',
					// react 18s `useId` break this rule
					'valid-id': 'off',
				},
			});
			expect(await axe(container)).toHaveNoViolations();
		});
	});
	describe('With isActive applied', () => {
		function createList(
			activeStatus: ProgressIndicatorItemStatus,
			extraProperties: Partial<ProgressIndicatorItem>
		): ProgressIndicatorProps {
			return {
				items: [
					{
						...extraProperties,
						label: 'Done',
						status: 'done',
						isActive: activeStatus === 'done',
					},
					{
						...extraProperties,
						label: 'Blocked',
						status: 'blocked',
						isActive: activeStatus === 'blocked',
					},
					{
						...extraProperties,
						label: 'Doing',
						status: 'doing',
						isActive: activeStatus === 'doing',
					},
					{
						...extraProperties,
						label: 'Error',
						status: 'error',
						isActive: activeStatus === 'error',
					},
					{
						...extraProperties,
						label: 'Saved',
						status: 'saved',
						isActive: activeStatus === 'saved',
					},
					{
						...extraProperties,
						label: 'Started',
						status: 'started',
						isActive: activeStatus === 'started',
					},
					{
						...extraProperties,
						label: 'To do',
						status: 'todo',
						isActive: activeStatus === 'todo',
					},
				],
			};
		}

		allProgressStatuses.forEach((status) => {
			const isWhiteListedStatus =
				activeProgressStatusAllowedList.includes(status);

			describe(`${status} status active`, () => {
				function runTest(
					extraProperties: Partial<ProgressIndicatorItem>
				): void {
					try {
						const { container } = renderProgressIndicator(
							createList(status, extraProperties)
						);
						if (isWhiteListedStatus) {
							expect(container).toMatchSnapshot();
						}
					} catch (err) {
						expect((err as Error).message).toBe(
							`The "${status}" status cannot be set to active. Please use one of the following status types instead: ${activeProgressStatusAllowedList.join(
								', '
							)}`
						);
					}
				}

				it('renders active link', () => {
					runTest({ href: '#' });
				});
				it('renders active button', () => {
					runTest({ onClick: console.log });
				});
			});
		});

		describe('renders valid HTML with no a11y violations', () => {
			it('renders valid active link', async () => {
				const { container } = renderProgressIndicator(
					createList('started', { href: '#' })
				);
				expect(container).toHTMLValidate({
					extends: ['html-validate:recommended'],
					rules: {
						'no-inline-style': 'off',
						// react 18s `useId` break this rule
						'valid-id': 'off',
					},
				});
				expect(await axe(container)).toHaveNoViolations();
			});
			it('renders valid active button', async () => {
				const { container } = renderProgressIndicator(
					createList('started', { onClick: console.log })
				);
				expect(container).toHTMLValidate({
					extends: ['html-validate:recommended'],
					rules: {
						'no-inline-style': 'off',
						// react 18s `useId` break this rule
						'valid-id': 'off',
					},
				});
				expect(await axe(container)).toHaveNoViolations();
			});
		});
	});
});
