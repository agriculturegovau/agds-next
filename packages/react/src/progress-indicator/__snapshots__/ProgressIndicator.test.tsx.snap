// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProgressIndicator With anchors renders correctly 1`] = `
<div>
  <section
    class="css-gkrtqa-boxStyles"
  >
    <div
      class="css-1upttg4-boxStyles"
    >
      <h2
        class="css-18vadwj-boxStyles-Text"
      >
        Progress
      </h2>
      <span
        class="css-1q2emlt-boxStyles-Text"
      >
        1 of 3 steps completed
      </span>
    </div>
    <button
      aria-controls="collapsing-side-bar-:r0:-body"
      aria-expanded="false"
      class="css-14aotnf-BaseButton-boxStyles-SideBarCollapseButton"
      id="collapsing-side-bar-:r0:-button"
      type="button"
    >
      1 of 3 steps completed
      <svg
        aria-hidden="true"
        class="css-1te2wbi-SideBarCollapseButton"
        clip-rule="evenodd"
        focusable="false"
        height="24"
        role="img"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <polyline
          points="6 9 12 15 18 9"
        />
      </svg>
    </button>
    <div
      class="css-1x3p89f-CollapsingSideBar"
      id="collapsing-side-bar-:r0:-body"
      style="display: none; height: 0px; overflow: hidden;"
    >
      <div
        class="css-1l5cwjh-boxStyles"
      >
        <ul
          class="css-ggltr2-boxStyles-ProgressIndicatorList"
        >
          <li
            class="css-jx6rss-boxStyles-ProgressIndicatorItem"
          >
            <a
              class="css-p21cw4-TextLink-boxStyles-ProgressIndicatorItem"
              href="#"
            >
              <span
                class="css-18cnah9-boxStyles"
              >
                <span
                  class="css-rz0ge3-ProgressIndicatorItemTimeline"
                  data-agds-progress-indicator-item-timeline-action=""
                />
                <span
                  class="css-yagt2a-ProgressIndicatorItemIcon"
                  data-agds-progress-indicator-item-ring=""
                >
                  <svg
                    aria-hidden="true"
                    class="css-4r8qyn-ProgressIndicatorItemIcon"
                    clip-rule="evenodd"
                    focusable="false"
                    height="24"
                    role="img"
                    viewBox="0 0 24 24"
                    width="24"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      clip-rule="evenodd"
                      d="M12 24C18.6274 24 24 18.6274 24 12C24 5.37258 18.6274 0 12 0C5.37258 0 0 5.37258 0 12C0 18.6274 5.37258 24 12 24ZM17.7071 9.70711C18.0976 9.31658 18.0976 8.68342 17.7071 8.29289C17.3166 7.90237 16.6834 7.90237 16.2929 8.29289L10 14.5858L7.70711 12.2929C7.31658 11.9024 6.68342 11.9024 6.29289 12.2929C5.90237 12.6834 5.90237 13.3166 6.29289 13.7071L9.29289 16.7071C9.68342 17.0976 10.3166 17.0976 10.7071 16.7071L17.7071 9.70711Z"
                      fill="currentColor"
                      fill-rule="evenodd"
                      stroke="none"
                    />
                  </svg>
                </span>
                <span
                  class="css-rz0ge3-ProgressIndicatorItemTimeline"
                  data-agds-progress-indicator-item-timeline-action=""
                />
              </span>
              <span
                class="css-1veqpy5-boxStyles"
                data-agds-progress-indicator-item-text-container=""
              >
                <span
                  class="css-7njy8a-boxStyles-Text"
                  data-agds-progress-indicator-item-text=""
                >
                  Introduction
                </span>
                <span
                  class="css-1sbzqg7-boxStyles-Text-ProgressIndicatorItem"
                >
                  Completed
                </span>
              </span>
            </a>
          </li>
          <li
            class="css-jx6rss-boxStyles-ProgressIndicatorItem"
          >
            <a
              class="css-1a1phxe-TextLink-boxStyles-ProgressIndicatorItem"
              href="#"
            >
              <span
                class="css-18cnah9-boxStyles"
              >
                <span
                  class="css-rz0ge3-ProgressIndicatorItemTimeline"
                  data-agds-progress-indicator-item-timeline-action=""
                />
                <span
                  class="css-liqiuw-ProgressIndicatorItemIcon"
                  data-agds-progress-indicator-item-ring=""
                >
                  <svg
                    aria-hidden="true"
                    class="css-jip0vh-ProgressIndicatorItemIcon"
                    clip-rule="evenodd"
                    focusable="false"
                    height="24"
                    role="img"
                    viewBox="0 0 24 24"
                    width="24"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M6.5 13.5C7.32843 13.5 8 12.8284 8 12C8 11.1716 7.32843 10.5 6.5 10.5C5.67157 10.5 5 11.1716 5 12C5 12.8284 5.67157 13.5 6.5 13.5Z"
                      fill="currentColor"
                      stroke="none"
                    />
                    <path
                      d="M12 13.5C12.8284 13.5 13.5 12.8284 13.5 12C13.5 11.1716 12.8284 10.5 12 10.5C11.1716 10.5 10.5 11.1716 10.5 12C10.5 12.8284 11.1716 13.5 12 13.5Z"
                      fill="currentColor"
                      stroke="none"
                    />
                    <path
                      d="M17.5 13.5C18.3284 13.5 19 12.8284 19 12C19 11.1716 18.3284 10.5 17.5 10.5C16.6716 10.5 16 11.1716 16 12C16 12.8284 16.6716 13.5 17.5 13.5Z"
                      fill="currentColor"
                      stroke="none"
                    />
                    <circle
                      cx="12"
                      cy="12"
                      r="10.5"
                    />
                  </svg>
                </span>
                <span
                  class="css-rz0ge3-ProgressIndicatorItemTimeline"
                  data-agds-progress-indicator-item-timeline-action=""
                />
              </span>
              <span
                class="css-82cgfm-boxStyles"
                data-agds-progress-indicator-item-text-container=""
              >
                <span
                  class="css-7njy8a-boxStyles-Text"
                  data-agds-progress-indicator-item-text=""
                >
                  Business Contacts
                </span>
                <span
                  class="css-1sbzqg7-boxStyles-Text-ProgressIndicatorItem"
                >
                  In progress
                </span>
              </span>
            </a>
          </li>
          <li
            class="css-jx6rss-boxStyles-ProgressIndicatorItem"
          >
            <a
              class="css-p21cw4-TextLink-boxStyles-ProgressIndicatorItem"
              href="#"
            >
              <span
                class="css-18cnah9-boxStyles"
              >
                <span
                  class="css-rz0ge3-ProgressIndicatorItemTimeline"
                  data-agds-progress-indicator-item-timeline-action=""
                />
                <span
                  class="css-yagt2a-ProgressIndicatorItemIcon"
                  data-agds-progress-indicator-item-ring=""
                >
                  <svg
                    aria-hidden="true"
                    class="css-cyyqnp-ProgressIndicatorItemIcon"
                    clip-rule="evenodd"
                    focusable="false"
                    height="24"
                    role="img"
                    viewBox="0 0 24 24"
                    width="24"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <circle
                      cx="12"
                      cy="12"
                      r="10.5"
                      stroke-dasharray="4 4"
                      stroke-linecap="round"
                    />
                  </svg>
                </span>
                <span
                  class="css-rz0ge3-ProgressIndicatorItemTimeline"
                  data-agds-progress-indicator-item-timeline-action=""
                />
              </span>
              <span
                class="css-1veqpy5-boxStyles"
                data-agds-progress-indicator-item-text-container=""
              >
                <span
                  class="css-7njy8a-boxStyles-Text"
                  data-agds-progress-indicator-item-text=""
                >
                  Case Studies
                </span>
                <span
                  class="css-1sbzqg7-boxStyles-Text-ProgressIndicatorItem"
                >
                  Not started
                </span>
              </span>
            </a>
          </li>
        </ul>
      </div>
    </div>
  </section>
</div>
`;

exports[`ProgressIndicator With buttons renders correctly 1`] = `
<div>
  <section
    class="css-gkrtqa-boxStyles"
  >
    <div
      class="css-1upttg4-boxStyles"
    >
      <h2
        class="css-18vadwj-boxStyles-Text"
      >
        Progress
      </h2>
      <span
        class="css-1q2emlt-boxStyles-Text"
      >
        1 of 3 steps completed
      </span>
    </div>
    <button
      aria-controls="collapsing-side-bar-:r2:-body"
      aria-expanded="false"
      class="css-14aotnf-BaseButton-boxStyles-SideBarCollapseButton"
      id="collapsing-side-bar-:r2:-button"
      type="button"
    >
      1 of 3 steps completed
      <svg
        aria-hidden="true"
        class="css-1te2wbi-SideBarCollapseButton"
        clip-rule="evenodd"
        focusable="false"
        height="24"
        role="img"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <polyline
          points="6 9 12 15 18 9"
        />
      </svg>
    </button>
    <div
      class="css-1x3p89f-CollapsingSideBar"
      id="collapsing-side-bar-:r2:-body"
      style="display: none; height: 0px; overflow: hidden;"
    >
      <div
        class="css-1l5cwjh-boxStyles"
      >
        <ul
          class="css-ggltr2-boxStyles-ProgressIndicatorList"
        >
          <li
            class="css-jx6rss-boxStyles-ProgressIndicatorItem"
          >
            <button
              class="css-ujttmb-BaseButton-boxStyles-ProgressIndicatorItem"
              type="button"
            >
              <span
                class="css-18cnah9-boxStyles"
              >
                <span
                  class="css-rz0ge3-ProgressIndicatorItemTimeline"
                  data-agds-progress-indicator-item-timeline-action=""
                />
                <span
                  class="css-yagt2a-ProgressIndicatorItemIcon"
                  data-agds-progress-indicator-item-ring=""
                >
                  <svg
                    aria-hidden="true"
                    class="css-4r8qyn-ProgressIndicatorItemIcon"
                    clip-rule="evenodd"
                    focusable="false"
                    height="24"
                    role="img"
                    viewBox="0 0 24 24"
                    width="24"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      clip-rule="evenodd"
                      d="M12 24C18.6274 24 24 18.6274 24 12C24 5.37258 18.6274 0 12 0C5.37258 0 0 5.37258 0 12C0 18.6274 5.37258 24 12 24ZM17.7071 9.70711C18.0976 9.31658 18.0976 8.68342 17.7071 8.29289C17.3166 7.90237 16.6834 7.90237 16.2929 8.29289L10 14.5858L7.70711 12.2929C7.31658 11.9024 6.68342 11.9024 6.29289 12.2929C5.90237 12.6834 5.90237 13.3166 6.29289 13.7071L9.29289 16.7071C9.68342 17.0976 10.3166 17.0976 10.7071 16.7071L17.7071 9.70711Z"
                      fill="currentColor"
                      fill-rule="evenodd"
                      stroke="none"
                    />
                  </svg>
                </span>
                <span
                  class="css-rz0ge3-ProgressIndicatorItemTimeline"
                  data-agds-progress-indicator-item-timeline-action=""
                />
              </span>
              <span
                class="css-1veqpy5-boxStyles"
                data-agds-progress-indicator-item-text-container=""
              >
                <span
                  class="css-7njy8a-boxStyles-Text"
                  data-agds-progress-indicator-item-text=""
                >
                  Introduction
                </span>
                <span
                  class="css-1sbzqg7-boxStyles-Text-ProgressIndicatorItem"
                >
                  Completed
                </span>
              </span>
            </button>
          </li>
          <li
            class="css-jx6rss-boxStyles-ProgressIndicatorItem"
          >
            <button
              class="css-1ydt447-BaseButton-boxStyles-ProgressIndicatorItem"
              type="button"
            >
              <span
                class="css-18cnah9-boxStyles"
              >
                <span
                  class="css-rz0ge3-ProgressIndicatorItemTimeline"
                  data-agds-progress-indicator-item-timeline-action=""
                />
                <span
                  class="css-liqiuw-ProgressIndicatorItemIcon"
                  data-agds-progress-indicator-item-ring=""
                >
                  <svg
                    aria-hidden="true"
                    class="css-jip0vh-ProgressIndicatorItemIcon"
                    clip-rule="evenodd"
                    focusable="false"
                    height="24"
                    role="img"
                    viewBox="0 0 24 24"
                    width="24"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M6.5 13.5C7.32843 13.5 8 12.8284 8 12C8 11.1716 7.32843 10.5 6.5 10.5C5.67157 10.5 5 11.1716 5 12C5 12.8284 5.67157 13.5 6.5 13.5Z"
                      fill="currentColor"
                      stroke="none"
                    />
                    <path
                      d="M12 13.5C12.8284 13.5 13.5 12.8284 13.5 12C13.5 11.1716 12.8284 10.5 12 10.5C11.1716 10.5 10.5 11.1716 10.5 12C10.5 12.8284 11.1716 13.5 12 13.5Z"
                      fill="currentColor"
                      stroke="none"
                    />
                    <path
                      d="M17.5 13.5C18.3284 13.5 19 12.8284 19 12C19 11.1716 18.3284 10.5 17.5 10.5C16.6716 10.5 16 11.1716 16 12C16 12.8284 16.6716 13.5 17.5 13.5Z"
                      fill="currentColor"
                      stroke="none"
                    />
                    <circle
                      cx="12"
                      cy="12"
                      r="10.5"
                    />
                  </svg>
                </span>
                <span
                  class="css-rz0ge3-ProgressIndicatorItemTimeline"
                  data-agds-progress-indicator-item-timeline-action=""
                />
              </span>
              <span
                class="css-82cgfm-boxStyles"
                data-agds-progress-indicator-item-text-container=""
              >
                <span
                  class="css-7njy8a-boxStyles-Text"
                  data-agds-progress-indicator-item-text=""
                >
                  Business Contacts
                </span>
                <span
                  class="css-1sbzqg7-boxStyles-Text-ProgressIndicatorItem"
                >
                  In progress
                </span>
              </span>
            </button>
          </li>
          <li
            class="css-jx6rss-boxStyles-ProgressIndicatorItem"
          >
            <button
              class="css-ujttmb-BaseButton-boxStyles-ProgressIndicatorItem"
              type="button"
            >
              <span
                class="css-18cnah9-boxStyles"
              >
                <span
                  class="css-rz0ge3-ProgressIndicatorItemTimeline"
                  data-agds-progress-indicator-item-timeline-action=""
                />
                <span
                  class="css-yagt2a-ProgressIndicatorItemIcon"
                  data-agds-progress-indicator-item-ring=""
                >
                  <svg
                    aria-hidden="true"
                    class="css-cyyqnp-ProgressIndicatorItemIcon"
                    clip-rule="evenodd"
                    focusable="false"
                    height="24"
                    role="img"
                    viewBox="0 0 24 24"
                    width="24"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <circle
                      cx="12"
                      cy="12"
                      r="10.5"
                      stroke-dasharray="4 4"
                      stroke-linecap="round"
                    />
                  </svg>
                </span>
                <span
                  class="css-rz0ge3-ProgressIndicatorItemTimeline"
                  data-agds-progress-indicator-item-timeline-action=""
                />
              </span>
              <span
                class="css-1veqpy5-boxStyles"
                data-agds-progress-indicator-item-text-container=""
              >
                <span
                  class="css-7njy8a-boxStyles-Text"
                  data-agds-progress-indicator-item-text=""
                >
                  Case Studies
                </span>
                <span
                  class="css-1sbzqg7-boxStyles-Text-ProgressIndicatorItem"
                >
                  Not started
                </span>
              </span>
            </button>
          </li>
        </ul>
      </div>
    </div>
  </section>
</div>
`;
